IRGen: Fix enable-testing of internal with resilient super class

aschwaighofer opened the pull request:
Enable testing makes `internal` types visible from outside the module.
We can no longer treat super classes as resilient.

Follow-up to #41044.

rdar://90489618

aschwaighofer committed:
IRGen: Fix enable-testing of internal with resilient super class

Enable testing makes `internal` types visible from outside the module.
We can no longer treat super classes as resilient.

Follow-up to #41044.

rdar://90489618

aschwaighofer commented:
@swift-ci Please test

aschwaighofer commented on lib/IRGen/GenClass.cpp:
```diff
@@ -229,11 +229,22 @@ namespace {
+                      /*useDC=*/nullptr,
+                      /*treatUsableFromInlineAsPublic=*/true)
+                  .isPublic());
+        if (IGM.hasResilientMetadata(superclassDecl,
```
We need to "pass down" the fact that the super class is not to be treated as resilient to the recursive call `addFieldsForClass` /`addClassMembers` on the super class.

slavapestov replied:
Right, maybe there's some other way, but this is the second bug we've fixed related to enable-testing vs class resilience, and it looks like both fixes added accidental complexity. We're just factoring the computations incorrectly here.

aschwaighofer commented:
@slavapestov I have changed the patch so that we are now additionally passing a `rootClass` parameter (the most derived sub class we are currently looking at) to the `isResilient...` predicates. And implemented the check inside the predicates.

aschwaighofer commented:
@swift-ci Please test
